docs/proposals/2017-11-17_-_Ratings.md

```diff
@@ -0,0 +1,117 @@
+# Problem
+
+A common genre of document across many publishers is a consumer review of products or media, such as movies, books, or gadgets. A consumer review is like any other article, but it often includes a formal rating of the item being reviewed. Rating values may be used:
+* to create renderable widgets on the site (Four gold star graphics and one gray star graphic for "4/5 stars") or
+* to sort content ("sort documents in this section by review score")
+
+There are further complications in a multi-tenant environment. Different publishers -- or even different desks or writers within a publisher -- may position the rating in different places within an article. Not all publishers or desks may use the same review system ("4/5 starts" vs "9.1 (out of 10)" vs "Recommended" or "Not Recommended")
+
+In some cases, like a round-up or a gift guide, a single article may also have multiple ratings for different items.
+
+This proposal attempts to address most of these cases by providing new content elements for these purposes.
+
+# Proposal
+
+Two new content elements that represent numeric and non-numeric ratings values.
+
+## New 'Rating' element
+
+A new content element `rating` will be added to indicate a numeric rating, with the following fields:
+
+* `rating` -- a float indicating the item's value (required)
+* `min` -- a float the minimum possible value of rating
+* `max` -- a float indicating the maximum possible value of rating
+* `units` -- a string describing the rating units (e.g., "stars," "thumbs")
+
+### Example
+```
+{
+  "type": "rating",
+  "rating": 4.5,
+  "min": 0,
+  "max": 5.0,
+  "units": "stars"
+}
+```
+
+## New 'Endorsement' element
+
+A new content element `endorsement` will be added to indicate text-only appraisals
+
+* `endorsement`: a string indicating the value (e.g. "Buy" or "Don't Buy") (required)
+
+
+### Example
+```
+{
+  "type": "endorsement",
+  "endorsement": "Recommended"
+}
+```
+
+`endorsements` should be used primarily when there is no quantifiable relationship between the various possible values.
+
+The set of possible values for the `units` and `endorsement` fields are left deliberately undefined here.  The only restrictions on `rating`, `min` and `max` are that `rating` is required and `min` < `rating` < `max`. Templates and App UIs may implement restrictions on new input as they see fit, but they *must be able to interpret any arbitrary values for these fields in a reasonable way.*
+
+
+## Document Example
+
+```
+{
+  "type": "story",
+  "version": "0.5.9",
+  "headlines": {
+    "basic": "Not So Beastly"
+  },
+  "content_elements": [
+    {
+        "type": "text",
+        "content": "Disney's <em>Beauty and the Beast</em> is pretty good, all things considered."
+    },
+    {
+        "type": "rating",
+        "rating": 4.5,
+        "min": 0,
+        "max": 5,
+        "units": "stars"
```

Comment:
Do we need to include this here, and if so, should it be displayable text?

Reply:
"Units" is mainly there to disambiguate different types of ratings... 

A single organization might rate movies n out of five stars, but rate lies as p out of four pinnochios.  So you could right a generic text renderer that just output ({rating} out of {max} {units}), or you might check specifically for "stars" and then go load a bunch of star pngs.  

As for whether or not to make it explicitly about reviews -- I'm not sure.  I think it's reasonable to use them for rating things that aren't *exactly* reviews, but I agree that they could be confused with other uses of the word "rating."  (And ironically, an age-appropriateness rating is better captured by "endorsement")

For lack of any better terms, I'm inclined to leave them as is, but I'd love to hear any suggestions